### PR TITLE
RegExp pick

### PR DESCRIPTION
**Pull request by tristaaan**
- can do pick with RegExp and strings.
``` javascript
var sampleObject = {a: 1, B: 2, C: 3, d: 4}
pick(sampleObject, RegExp('[A-Z]')); // {B:2, C:3}
```
- added `-t 100` to package.json to ensure 100% test coverage

### CODE REVIEW

**Comment by tjmehta** [comment]
Cool, I like this. Thanks @tristaaan will pull it tonight.

**Comment by tjmehta** [comment]
Could you update the Readme as well?

**Comment by stoeffel** [comment]
@tristaaan you should update the jsdoc in `pick.js` too.

**Comment by tristaaan** [comment]
Updated readme and jsdoc. It was missing some minor test cases too. (was assuming an array's elements were all of the same type, either regexp or string)

File: pick.js [hunk 1]
```diff
@@ -27,18 +27,49 @@ module.exports = function () {
...
   return function (key) {
     if (key in from) {
       to[key] = from[key];
     }
   };
+}
+
+function copyWithRegExp (from, to, regexp) {
+  return function (key) {
+    if (regexp.test(key) && !(key in to)) {
```
Comment (tjmehta) [inline pick.js]: Pick should always override `to` (it should not work like `101/default`)

**Comment by tjmehta** [comment]
Thanks for the promptness. I am going to sleep on this, one more night. I think I have some ideas on how to extend pick's functionality and I want to make sure what you've done here will play nice. 

Idea: Support key-mapping functionality by supplying a hash of keys to function (similar to what is mentioned here: https://github.com/tjmehta/101/issues/49#issuecomment-73480023).

**Comment by tjmehta** [comment]
After thinking about this further I believe the RegExp functionality makes sense. 
I like the cleaner implementation. 

Just waiting on moving `isRegExp` to `101/is-regexp`. 

Everything else here LGTM. :+1:

**Comment by tristaaan** [comment]
`isRegExp` and tests was added by @stoeffel. Will be useful for match function #5.

**Comment by stoeffel** [comment]
> Just waiting on moving isRegExp to 101/is-regexp.

done

**Comment by tjmehta** [comment]
Thanks! :100:

**Comment by tjmehta** [comment]
Published in v0.11.0

**Pull request merged**